smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatManager.java

```diff
@@ -199,6 +206,40 @@ public void processStanza(Stanza packet) {
         };
         connection.addAsyncStanzaListener(invitationPacketListener, INVITATION_FILTER);
 
+        // Listens for all messages that include an XEP-0249 GroupChatInvitation extension and fire the invitation
+        // listeners
+        StanzaListener directInvitationPacketListener = new StanzaListener() {
+            @Override
+            public void processStanza(Stanza packet) {
+                final Message message = (Message) packet;
+                GroupChatInvitation invite =
+                    packet.getExtension(GroupChatInvitation.class);
+
+                if (invite.getRoomAddress() != null) {
```

Comment:
This check is unnecessary, as it is impossible to create a `GroupChatInvitation` that hasn't set a room address.